abodepy/event_controller.py

```diff
@@ -46,6 +50,69 @@ def stop(self):
         """Tell the subscription thread to terminate - will block."""
         self._socketio.stop()
 
+    def add_connection_status_callback(self, devices, callback):
+        """Register callback for Abode server connection status."""
+        if not devices:
+            return False
+
+        if not isinstance(devices, (tuple, list)):
+            devices = [devices]
+
+        for device in devices:
+            device_id = device
+
+            if isinstance(device, AbodeDevice):
+                device_id = device.device_id
+
+                if not self._abode.get_device(device_id):
+                    raise AbodeException((ERROR.EVENT_DEVICE_INVALID))
+
+            if isinstance(device, AbodeAutomation):
+                device_id = device.automation_id
+
+                if not self._abode.get_automation(device_id):
+                    raise AbodeException((ERROR.EVENT_DEVICE_INVALID))
+
+            _LOGGER.debug(
+                "Subscribing to Abode connection updates for: %s", device_id)
+
+        self._connection_status_callbacks[device_id].append((callback))
+
+        return True
+
+    def remove_connection_status_callback(self, devices):
+        """Unregister connection status callbacks."""
+        if not devices:
+            return False
+
+        if not isinstance(devices, (tuple, list)):
+            devices = [devices]
+
+        for device in devices:
+            device_id = device
+
+            if isinstance(device, AbodeDevice):
+                device_id = device.device_id
+
+                if not self._abode.get_device(device_id):
+                    raise AbodeException((ERROR.EVENT_DEVICE_INVALID))
+
+            if isinstance(device, AbodeAutomation):
+                device_id = device.automation_id
+
+                if not self._abode.get_automation(device_id):
+                    raise AbodeException((ERROR.EVENT_DEVICE_INVALID))
+
+            if device_id not in self._connection_status_callbacks:
+                return False
+
+            _LOGGER.debug(
+                "Unsubscribing from Abode connection updates for : %s", device_id)
+
+            self._connection_status_callbacks[device_id].clear()
```

Comment:
Small problem here. The Abode multi sensor is one physical device but acts as three separate entities (humidity, lux and temp). This means the device has three callback methods:

`{'RF: 01288e80': [<bound method AbodeEntity._update_connection_status of <Entity Office Sensor Temperature: 71.0>>, <bound method AbodeEntity._update_connection_status of <Entity Office Sensor Lux: 1.0>>]}`

When a single entity is disabled, `clear()` is called on `self._connection_status_callbacks[device_id]` which deletes all three callbacks when it should only be deleting one. I'm not sure how to choose a specific callback in this list...

Reply:
Fixed this issue by using the `unique_id` property instead of `device_id`. This way each entity of the multi sensor gets its own key in the callback dict.